src/Utilities/ConstantExpressions.hpp

```diff
@@ -207,15 +208,36 @@ constexpr T min_by_magnitude(std::initializer_list<T> ilist) {
 }
 //@}
 
+/// \ingroup ConstantExpressionsGroup
+/// \brief Clamps the value between lo and hi
+///
+/// If v compares less than lo, returns lo; otherwise if hi compares less than
+///  v, returns hi; otherwise returns v.
+//@{
+template <class T, class Compare>
+constexpr const T& clamp(const T& v, const T& lo, const T& hi, Compare comp) {
+  // reason for NOLINT:
+  // "warning: do not implicitly decay an array into a pointer; consider using
```

Comment:
why is this warning occuring? we can't find any instances of clamp being used with an array